kubetools/dev/backends/docker_compose/docker_util.py

```diff
@@ -136,8 +136,14 @@ def get_containers_status(
         if not env:
             env = compose_project.replace(docker_name, '')
 
-        # Where the name is compose-name_container_N, get container
-        name = container.name.split('_')[1]
+        # if the container was made in v1, it will be composename_container_N
+        converted_name = container.name.replace('_', '-')
+
+        # we need to keep the middle part of the name
+        # for example if we have a container called `composename-container-1-N`
+        # we want to keep `container-1`
+        middle_names = converted_name.split('-')[1:-1]
+        name = '-'.join(middle_name for middle_name in middle_names)
```

Comment:
Can you please add tests for this? The main goal being to document the behaviour of this logic and its outcome when applied in each of the possible situations.

Note that the added test "`complex_named_app`" doesn't cover `docker-compose` code. These tests only cover `kubernetes` config generation.

Reply:
I've factored this bit of the code out and added tests for different scenarios, plus added test coverage for the `dockerise_label()` function on which this is heavily reliant:
 [b9cc5a4](https://github.com/EDITD/kubetools/pull/163/commits/b9cc5a4cefc1758b48f11241471835c62c3ce99f)